pkg/ddc/alluxio/engine.go

```diff
@@ -73,6 +76,13 @@ func Build(id string, ctx cruntime.ReconcileRequestContext) (base.Engine, error)
 		return nil, fmt.Errorf("engine %s is failed to parse", ctx.Name)
 	}
 
+	// Get access mode
+	dataset, err := utils.GetDataset(ctx.Client, ctx.Name, ctx.Namespace)
+	if err != nil {
+		return nil, err
+	}
+	engine.AccessModes = dataset.Spec.AccessModes
+
```

Comment:
The `Build` func may be called before creation of the Dataset because according to the logic of `ReconcileInternal()`, `GetOrCreateEngine()` ([runtime_controller.go](https://github.com/fluid-cloudnative/fluid/blob/master/pkg/controllers/runtime_controller.go#L70)) will be called before the existence check for the Dataset. So a error may be thrown here